Protecs/src/test/java/uk/ac/ucl/protecs/sim/WorkplaceTesting.java

```diff
@@ -0,0 +1,220 @@
+package uk.ac.ucl.protecs.sim;
+
+import uk.ac.ucl.protecs.helperFunctions.*;
+import uk.ac.ucl.protecs.objects.Household;
+import uk.ac.ucl.protecs.objects.Person;
+import uk.ac.ucl.protecs.objects.Workplace;
+import uk.ac.ucl.protecs.objects.Location.LocationCategory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class WorkplaceTesting{
+	// ================================================ Testing =======================================================================
+	// ===== Here we test that the model is reading in workplaces from the census csv file and are being stored as workplace objects. =
+	// ===== We check that subsequently the workplace bubbles are created and are associated with the workplace location created. =====
+	// ===== We check that the bubbles are then created for each person, and that those who share a workplace have everyone associated=
+	// ===== with that workplace has been included in the bubble.                                                               =======
+	// ===== We check that people travel to their workplace location in the model.                                              =======
+	// ===== We check that the parameters being used to predict workplace contacts are being loaded                             =======
+	// ================================================================================================================================
+	private final String params;
+	
+	public WorkplaceTesting(String fileName) {
+		this.params = fileName;
+	}
+
+	@Test
+	public void checkPopulationWorkplacesAreBeingLoaded() {
+		// workplaces are read in from the csv file now, initialise the simulation
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// check that the workplace locations have been created
```

Comment:
please add a check to see if sim.workplaces is null, lest it torpedo things

Reply:
Best way to do this is to just assert not null!